test/foundry/dapp/Application.t.sol

```diff
@@ -426,68 +361,161 @@ contract ApplicationTest is ERC165Test {
+            _tokenId,
+            _initialSupply,
+            ""
+        );
 
-        uint256 recipientBalance = _erc20Token.balanceOf(address(_recipient));
-        uint256 appBalance = _erc20Token.balanceOf(address(_appContract));
+        uint256 recipientBalance = _erc1155SingleToken.balanceOf(
+            _recipient,
+            _tokenId
+        );
+        uint256 appBalance = _erc1155SingleToken.balanceOf(
+            address(_appContract),
+            _tokenId
+        );
 
         _expectEmitOutputExecuted(output, proof);
         _appContract.executeOutput(output, proof);
 
         assertEq(
-            _erc20Token.balanceOf(address(_recipient)),
+            _erc1155SingleToken.balanceOf(address(_appContract), _tokenId),
+            appBalance - _transferAmount,
+            "Application contract should have the transfer amount deducted"
+        );
+        assertEq(
+            _erc1155SingleToken.balanceOf(_recipient, _tokenId),
             recipientBalance + _transferAmount,
             "Recipient should have received the transfer amount"
         );
 
-        assertEq(
-            _erc20Token.balanceOf(address(_appContract)),
-            appBalance - _transferAmount,
-            "Application contract should have the transfer amount deducted"
+        assertTrue(
+            _wasOutputExecuted(proof),
+            "Output should be marked as executed"
         );
 
+        _expectRevertOutputNotReexecutable(output);
+        _appContract.executeOutput(output, proof);
+    }
+
+    function testERC1155BatchTransferToENS() external {
+        string memory name = "ERC1155BatchToENSDelegateCallVoucher";
+        bytes memory output = _getOutput(name);
+        OutputValidityProof memory proof = _getProof(name);
+
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                IERC1155Errors.ERC1155InsufficientBalance.selector,
+                address(_appContract),
+                0,
+                _transferAmounts[0],
+                _tokenIds[0]
+            )
+        );
+        _appContract.executeOutput(output, proof);
+
+        vm.prank(_tokenOwner);
+        _erc1155BatchToken.safeBatchTransferFrom(
+            _tokenOwner,
+            address(_appContract),
+            _tokenIds,
+            _initialSupplies,
+            ""
+        );
+
+        _expectEmitOutputExecuted(output, proof);
+        _appContract.executeOutput(output, proof);
+
+        for (uint256 i; i < _tokenIds.length; ++i) {
+            assertEq(
+                _erc1155BatchToken.balanceOf(
+                    address(_appContract),
+                    _tokenIds[i]
+                ),
+                _initialSupplies[i] - _transferAmounts[i],
+                "Application contract should have the transfer amount deducted"
+            );
+            assertEq(
+                _erc1155BatchToken.balanceOf(_recipient, _tokenIds[i]),
+                _transferAmounts[i],
```

Comment:
I know that we _know_ that the application contract starts with `_initialSupplies[i]` and the recipient with zero tokens for each token type, but it would be better to construct `appBalances` and `recipientBalances` arrays before the transfer, and use them for the checks after the transfer.